src/transaction-types.ts

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-restricted-globals */
+import { Hex } from './hex';
+
+export type Transaction = (LegacyTx | EIP2930Tx | EIP1559Tx) & Signature;
+
+export type Signature = {
+  /**
+   * EC signature parameter
+   * 32 bytes long sequence.
+   */
+  r?: bigint | string | number | Buffer;
+
+  /**
+   * EC signature parameter
+   * Signature proof.
+   * 32 bytes long sequence
+   */
+  s?: bigint | string | number | Buffer;
+
+  /**
+   * Recovery identifier. It can be either 0x1b or 0x1c
+   * 1 byte long sequence
+   */
+  v?: bigint | string | number | Buffer;
+};
+
+/**
+ * Ethereum Legacy Transaction (Type-0)
```

Comment:
Legacy transactions technically don't have a transaction type.
```suggestion
 * Ethereum Legacy Transaction
```

Reply:
[e0557ff](https://github.com/MetaMask/utils/pull/74/commits/e0557ff2caa28552c155b8381d95611054888f34)